azure-auth-helper/src/main/java/com/microsoft/azure/auth/QueryStringUtil.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.auth;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class QueryStringUtil {
```

Comment:
Please do not implement such utility functions yourself. Security bugs are mostly from such implementations. Please use an existing library instead.

Reply:
will use URLEncodedUtils to parse